Bottom: 9594060050577ddc6969cde7443e53757fedf86c
Top:    e35f4d7d055454aad2db9f47ed70ef1c03cc8aed
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-08 20:11:21 +0000

Refresh of more-work-on-trying-to-speed-u

---

diff --git a/src/polygon1.c b/src/polygon1.c
index 7803be1..d958806 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -873,8 +873,7 @@ M_POLYAREA_intersect2 (jmp_buf * e, POLYAREA * afst, POLYAREA * bfst, int add)
     {
       do
 	{
-	  if (a->contours &&
-              a->contours->xmax >= b->contours->xmin &&
+	  if (a->contours->xmax >= b->contours->xmin &&
 	      a->contours->ymax >= b->contours->ymin &&
 	      a->contours->xmin <= b->contours->xmax &&
 	      a->contours->ymin <= b->contours->ymax)
@@ -947,7 +946,6 @@ M_POLYAREA_intersect (jmp_buf * e, POLYAREA * afst, POLYAREA * bfst, int add)
   while (add && (a = a->f) != afst);
 }				/* M_POLYAREA_intersect */
 
-
 static inline int
 cntrbox_inside (PLINE * c1, PLINE * c2)
 {
